dora/core/client/fs/src/main/java/alluxio/client/file/dora/WorkerLocationPolicy.java

```diff
@@ -39,17 +40,17 @@ public interface WorkerLocationPolicy {
    * This method should return exactly #{count} different workers, no more no less.
    * If the specified number of workers cannot be found, this method will throw
    * a {@link ResourceExhaustedException}.
-   *
+   * <p>
    * We want the semantics here to be explicit when the requirement cannot be satisfied.
    * So the caller should define its own logic handling the exception and finding backups.
    *
-   * @param blockWorkerInfos
+   * @param workers
    * @param fileId
    * @param count
    * @return a list of preferred workers
    * @throws ResourceExhaustedException if unable to return exactly #{count} workers
    */
-  List<BlockWorkerInfo> getPreferredWorkers(List<BlockWorkerInfo> blockWorkerInfos,
+  List<BlockWorkerInfo> getPreferredWorkers(WorkerClusterView workers,
```

Comment:
the return type is unchanged because the order of the workers returned is important.